Refactor Mass Order Cancel code to use Interface

AnshuMishra17 opened the pull request:
### Description
I have observed that MassAction Cancel is using the collection for orders cancel, whereas Order Cancel (order cancel from order edit section) is using Interface to put the order on hold.
So, I have refactor the Mass Order Cancel code to use Sales Order Interface.

### Manual testing scenarios
1. Go to Sales Order Grid
2. Select any number of orders to cancel
3. Select Cancel from Action Dropdown

AnshuMishra17 committed:
Refactor Mass Order Cancel code to use Interface

I have observed that MassAction Cancel is using the collection for orders cancel, whereas Order Cancel (order cancel from order edit section) is using Interface to put the order on hold.
So, I have refactor the Mass Order Cancel code to use Sales Order Interface.

sidolov commented on app/code/Magento/Sales/Controller/Adminhtml/Order/MassCancel.php:
```diff
@@ -9,23 +9,31 @@
+     * @param OrderManagementInterface $orderManagement
      */
-    public function __construct(Context $context, Filter $filter, CollectionFactory $collectionFactory)
+    public function __construct(Context $context, Filter $filter, CollectionFactory $collectionFactory, OrderManagementInterface $orderManagement)
```
Please, add new dependencies according to our [Backward Compatible Development Guide](http://devdocs.magento.com/guides/v2.2/contributor-guide/backward-compatible-development/#adding-a-constructor-parameter)

AnshuMishra17 replied:
Done

sidolov commented on app/code/Magento/Sales/Controller/Adminhtml/Order/MassCancel.php:
```diff
@@ -9,23 +9,31 @@
+    /**
+     * @var OrderManagementInterface
+     */
+    protected $orderManagement;
```
Please, make this property private

sidolov reviewed (changes requested):
@AnshuMishra17 , please, fix all issue according to review and take a look to failed tests

AnshuMishra17 committed:
Added backward Compatibility

Added backward compatibility for the new orderManagementInterface dependency and made the variable $orderManagement private.

Commit:
magento/magento2#14630 Fixed test failures

sidolov commented:
Hi @AnshuMishra17 , I fixed test failures

magento-engcom-team commented:
@AnshuMishra17 thank you for contributing. Please accept Community Contributors team invitation [here](https://github.com/orgs/magento/invitation) to gain extended permissions for this repository.

magento-engcom-team commented:
Hi @AnshuMishra17. Thank you for your contribution. 
Changes from your Pull Request will be available with the upcoming 2.3.0 release.
